common/interactive-object/server.mjs

```diff
@@ -297,9 +300,10 @@ class InteractiveObjectServerBaseClass {
     if (Array.isArray(dialogs)) d = randomChoice(dialogs);
 
     const result = await this.helper.callS2cAPI(playerID, 'dialog', 'showDialogWithPrompt', 60*1000, this.objectName, d, buttonText);
-    console.log(result);
     if (result.msg) {
-      this.editedFSM.states[nextState].kwargs.dialogs = result.msg;
+      if (this.editableDialogs) {
+        this.editableDialogs[this.config.FSM.states[nextState].kwargs.dialogs] = result.msg;
```

Comment:
Consider allowing the json file to specify the which variable name or state to fetch from. i.e.
```
let varName = '';
if (typeof kwargs.varName === 'string') varName = kwargs.varName;
else if (typeof kwargs.editState === 'string' ) varName = this.config.FSM.states[kwargs.editState].kwargs.dialogsVar;
else varName = this.config.FSM.states[nextState].kwargs.dialogsVar;
// ...
this.globalVars[varName] = result.msg;
```